### PR TITLE
[clang-tidy] Ignore deleted ctor in `bugprone-forwarding-reference-overload`

### LINKED ISSUE #88128: clang-tidy `bugprone-forwarding-reference-overload` false positive on `= delete;` constructor
The `bugprone-forwarding-reference-overload` check in clang-tidy will warn even for a `= delete;` constructor:

```cpp
// main.cpp
class Foo {
public:
  // ...

  template <typename T>
  Foo(T&&) = delete;

  // ...
};
```

```bash
clang-tidy -checks=bugprone-* main.cpp

/path/to/main.cpp:5:3: warning: constructor accepting a forwarding reference can hide the copy and move constructors [bugprone-forwarding-reference-overload]
    5 |   Foo(T&&) = delete;
      |   ^
```

In our case, we had various templated constructors and it looks like the original author decided to explicitly delete this to warn/prevent somebody making this mistake in the future.

```
$ clang-tidy --version
LLVM (http://llvm.org/):
  LLVM version 17.0.6
  Optimized build.
```

### PR DESCRIPTION
**Pull request by MikeWeller**
Fix `bugprone-forwarding-reference-overload` so it doesn't report a constructor that is deleted.

### CODE REVIEW

**Comment by MikeWeller** [comment]
https://github.com/llvm/llvm-project/issues/88128

File: clang-tools-extra/test/clang-tidy/checkers/bugprone/forwarding-reference-overload.cpp
Comment (MikeWeller) [inline clang-tools-extra/test/clang-tidy/checkers/bugprone/forwarding-reference-overload.cpp:259]: Although this is enough to trigger (or not after the fix) the issue, perhaps I should include a copy+move ctor below for clarity?

**Comment by EugeneZelenko** [comment]
Please mention changes in Release Notes.

**Review by PiotrZSL — COMMENTED** [review]
Missing release notes entry, except that looks fine.

**Comment by MikeWeller** [comment]
Sorry, didn't realize there are release notes I can updated in the change itself. Will do this Monday.

**Comment by MikeWeller** [comment]
Given the small size of the change I just did a rebase and force push to resolve conflicts with the release notes.

File: clang-tools-extra/docs/ReleaseNotes.rst
Comment (PiotrZSL) [inline clang-tools-extra/docs/ReleaseNotes.rst:105]: Not here, under `Changes in existing checks` section and follow current format there.

**Comment by MikeWeller** [comment]
@PiotrZSL apologies, have put it in the correct location. Is there a way to preview/check the `:doc:` thing? I assume there is some kind of preprocessor/tool step for this? I copied what the rest are doing and put the appropriate section of the https://clang.llvm.org/extra/clang-tidy/checks/bugprone/forwarding-reference-overload.html URL.

File: clang-tools-extra/docs/ReleaseNotes.rst
Comment (PiotrZSL) [inline clang-tools-extra/docs/ReleaseNotes.rst:289]: move it to line 149, to keep checks in alphabetical order.

**Review by HerrCai0907 — APPROVED** [review]
LGTM in source code part.

**Review by PiotrZSL — APPROVED** [review]

**Comment by MikeWeller** [comment]
FYI I don't have merge privs, not sure if I need to ping anybody. Then again I still see "This workflow requires approval from a maintainer", possibly because this is my first contribution?

File: clang-tools-extra/docs/ReleaseNotes.rst
Comment (MikeWeller) [reply clang-tools-extra/docs/ReleaseNotes.rst]: Done in bab66e1, let me know if you have a better suggestion

**Comment by HerrCai0907** [comment]
please fix format issue.`git-clang-format` is helpful.

**Pull request merged**
